lib/block-patterns.php

```diff
@@ -236,15 +236,52 @@ function load_remote_patterns() {
 	}
 }
 
+/**
+ * Register `Featured` (category) patterns from wordpress.org/patterns.
+ *
+ * @since 5.9.0
+ */
+function gutenberg_load_remote_featured_patterns() {
+	/**
+	 * Filter to disable remote block patterns.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool $should_load_remote
+	 */
+	$should_load_remote = apply_filters( 'should_load_remote_block_patterns', true );
+
+	if ( ! $should_load_remote ) {
+		return;
+	}
+
+	if ( ! WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'featured' ) ) {
+		register_block_pattern_category( 'featured', array( 'label' => __( 'Featured', 'gutenberg' ) ) );
+	}
+	$request             = new WP_REST_Request( 'GET', '/wp/v2/pattern-directory/patterns' );
+	$request['category'] = 26; // This is the `Featured` category id from pattern directory.
+	$response            = rest_do_request( $request );
+	if ( $response->is_error() ) {
+		return;
+	}
+	$patterns = $response->get_data();
+	foreach ( $patterns as $pattern ) {
+		$pattern_name = sanitize_title( $pattern['title'] );
+		if ( ! WP_Block_Patterns_Registry::get_instance()->is_registered( $pattern_name ) ) {
+			register_block_pattern( $pattern_name, (array) $pattern );
+		};
```

Comment:
I'm noticing some things that are slightly tangent to this PR, but I'll still mention them:

* As in the previous function, the pattern's title is sanitised but the rest of the payload isn't, neither here nor in the patterns registry class. Seems like a code smell to me — either we are oversanitising or undersanitising.
* Though it's not the only case in WP, I've always found it very odd that the backend should make a HTTP request to itself. :) I don't know what arguments support this against just querying directly.

Reply:
Good questions Miguel. @ryelle does the sanitisation has to do with the current way the patterns are included in the directory? The inclusion happens manually for now so it's `sanitised` by 'us'?